src/test/scala/lenthall/config/ScalaConfigSpec.scala

```diff
@@ -23,6 +25,7 @@ class ScalaConfigSpec extends FlatSpec with Matchers {
   it should "return the config value as options when present" in {
     exampleConfig.getConfigOption("configVal") should be('defined)
     exampleConfig.getStringOption("stringVal") should be(Some("string"))
+    exampleConfig.getPathOption("stringVal") should be(Some(Paths.get("string")))
```

Comment:
I'm philosophically curious about why we have `getNonPrimitiveType` in here. Is it really that much better than using `getStringOption map { Paths.get(_) }` in code?

Maybe a typeclass-y `getConstructibleFromPrimitiveOption[T]` :) (just kidding. Probably)